### ISSUE TITLE
VTKIO compression not working?

### ISSUE THREAD

**Issue by bzjan** [libMesh/libmesh#2434]
I am trying to save a solution on a mesh as a vtk file with compression. My problem is that the compression setting does not have an effect on the filesize.

I am using vtk7.1.

Here is my code:

```
#include "libmesh/libmesh.h"
#include "libmesh/mesh.h"							// libMesh::Mesh, mesh()
#include "libmesh/node.h"							// libMesh::Node
#include "libmesh/elem.h"							// libMesh::Elem
#include "libmesh/mesh_generation.h"
#include "libmesh/linear_implicit_system.h"
#include "libmesh/equation_systems.h"
#include "libmesh/numeric_vector.h"
#include "libmesh/dense_vector_base.h"
#include "libmesh/utility.h"
#include "libmesh/vtk_io.h"							// libMesh::VTKIO

using namespace libMesh;


int main (int argc, char** argv){
	
	// mesh
	LibMeshInit init(argc, argv);
	Mesh mesh(init.comm(), 2);
	MeshTools::Generation::build_square(mesh,  10, 10, -1., 1., -1., 1., QUAD9);
	
	// equations
	EquationSystems equation_systems(mesh);
	equation_systems.add_system<LinearImplicitSystem> ("Test");
	System& system = equation_systems.get_system("Test");
	system.add_variable("u", SECOND);
	equation_systems.init();
	
	// mesh output
	char filename[100];
	sprintf(filename,"compression_test_output.pvtu");
	libMesh::VTKIO vtkio(mesh);
	vtkio.set_compression(true);
	vtkio.write_equation_systems(filename,equation_systems);
	
	sprintf(filename,"no_compression_test_output.pvtu");
	vtkio.set_compression(false);
	vtkio.write_equation_systems(filename,equation_systems);
	
	return 0;
}
```

**Comment by jwpeterson** [comment]
I'm not very familiar with the VTK APIs, it's entirely possible we are using it wrong. All we do currently is:
```
if (this->_compress)
  writer->SetCompressorTypeToZLib();

writer->Write();
```
so you may have better luck asking about this on the VTK lists... and if you come up with a simple fix that works for you, it would be great if you could make a PR!

**Comment by bzjan** [comment]
Thanks for the pointer!

All that needs to be done is, add the following line: 
`writer->SetDataModeToBinary();`

Now I have working binary file output.

**Comment by jwpeterson** [comment]
Fixed in #2436.
